docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-enable-cluster-label-verification/_index.md

```diff
@@ -0,0 +1,96 @@
+
+## How The Migration Solve The Issue
+
+Non-Mimir component can merge with Mimir memberlist because by default, memberlist is not namespaced and applies globally. As an example, consider a Loki pod is terminated and the IP address reused by a new Mimir pod in the same cluster. This might cause other Loki components try to talk and sending data to this new Mimir pod.
+
+Cluster label will prevent such situation by only allowing communication for components that has same cluster label. Once that is enabled, before memberlist try to communicate with other components, it will verify whether that new components are having same cluster label and only allowing memberlist to communicate if the parties are having the same cluster label.
+
+## Migration
+
+### 1. Disable memberlist cluster label verification
+
+Cluster label verification flag is enabled by default. However, cluster label default value is an empty string. If different systems that using memberlist and not setting the cluster label, they can talk with each other. To disable cluster label verification flag, set the following structured config.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: true
+```
+
+Rollout the installation to apply the configuration changes by running `helm upgrade mimir-distributed -f values.yaml`.
+
+### 2. Set cluster label to all Mimir components
+
+Set cluster label to all Mimir components by setting the following configuration. Once the configuration is applied, all Mimir components will only communicate via memberlist if the other component is having the same cluster label.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: true
+      cluster_label: "$helm-release-name"
+```
+
+Apply the configuration changes by running `helm upgrade mimir-distributed -f values.yaml`.
+
+### 3. Enable memberlist cluster label verification
+
+Set `memberlist.cluster_label_verification_disabled` to false, to re-enable memberlist cluster label verification.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: false
+      cluster_label: "$helm-release-name"
+```
+
+Apply the configuration changes by running `helm upgrade mimir-distributed -f values.yaml`.
+
+## Verifying The Migration
+
+Once the rollout has been done run the following helm command to verify if we have successfully migrated the Mimir memberlist to check the cluster label.
+
+```
+helm --kube-context=[your-k8s-context] --namespace=[your-mimir-namespace] get values [your-mimir-release-name]
+```
+
+You should see the following config the in the values as what we have set above.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: false
+      cluster_label: "$helm-release-name"
+```
```

Comment:
Let's get folks to visit the `/memberlist` endpoint on a couple of pods to make sure they have the same view of all members. I think that would be more convincing verification because it means components are actually part of the same memberlist cluster.

Reply:
Didn't think about `/memberlist` endpoint. Yes that is make more sense to use this for the verification instead of just printing out back the values.yaml.